diff --git a/build/core/binary.mk b/build/core/binary.mk
index bc85d0b..08ae04f 100644
--- a/build/core/binary.mk
+++ b/build/core/binary.mk
@@ -487,6 +487,11 @@ all_objects := \
 	$(proto_generated_objects) \
 	$(addprefix $(TOPDIR)$(LOCAL_PATH)/,$(LOCAL_PREBUILT_OBJ_FILES))
 
+## Allow a device's own headers to take precedence over global ones
+ifneq ($(TARGET_SPECIFIC_HEADER_PATH),)
+LOCAL_C_INCLUDES := $(TOPDIR)$(TARGET_SPECIFIC_HEADER_PATH) $(LOCAL_C_INCLUDES)
+endif
+
 LOCAL_C_INCLUDES += $(TOPDIR)$(LOCAL_PATH) $(intermediates) $(base_intermediates)
 
 ifndef LOCAL_NDK_VERSION
