diff --git a/frameworks/base/libs/utils/Android.mk b/frameworks/base/libs/utils/Android.mk
index 9e43538..475c3ce 100644
--- a/frameworks/base/libs/utils/Android.mk
+++ b/frameworks/base/libs/utils/Android.mk
@@ -61,7 +61,7 @@ LOCAL_SRC_FILES:= $(commonSources)
 
 LOCAL_MODULE:= libutils
 
-LOCAL_CFLAGS += -DLIBUTILS_NATIVE=1 $(TOOL_CFLAGS)
+LOCAL_CFLAGS += -DLIBUTILS_NATIVE=1 $(TOOL_CFLAGS) -fpermissive
 LOCAL_C_INCLUDES += external/zlib
 
 ifeq ($(HOST_OS),windows)
